numpy/linalg/tests/test_linalg.py

```diff
@@ -845,6 +846,111 @@ def test_empty(self):
         assert_equal(norm(array([], dtype=self.dt)), 0.0)
         assert_equal(norm(atleast_2d(array([], dtype=self.dt))), 0.0)
 
+    def test_vector_types(self):
```

Comment:
Not sure how I would merge with matrices. There are some types that are explicitly forbidden for matrix types. Also, they don't all have the same kinds of norms. Even if they do all the same norm, there definitions are not always the same.

I suppose we could put them all in the same function, but it is not as clear as seeing a test involving vectors failing or one with matrices failing.